Grinnell-Events-iOS/Grinnell-Events-iOS/ViewController.m

```diff
@@ -75,33 +75,65 @@ - (void)viewDidLoad
             
             [self filterContentForSearchText:@"" scope:
              [[self.searchDisplayController.searchBar scopeButtonTitles] objectAtIndex:[self.searchDisplayController.searchBar selectedScopeButtonIndex]]];
             
             [self.tableView reloadData];
-            
+            //set dateformatter in order to convert string to date
+            NSString *dateString = self.sortedDateKeys.firstObject;
+            NSDateFormatter *dateFormatter = [[NSDateFormatter alloc] init];
+            dateFormatter.dateFormat = @"MM/dd/yyyy";
+            NSDate *firstDate = [dateFormatter dateFromString:dateString];
             // Set start and end dates in dayPicker
-            NSDate *firstDate = [NSDate dateFromString: self.sortedDateKeys.firstObject ];
+            //commented out this line of code because the date format was not set correctly
+            //NSDate *firstDate = [NSDate dateFromString: self.sortedDateKeys.firstObject ];
+            NSLog(@"First date is %@", firstDate.description);
             NSDate *lastDate = [NSDate dateFromString:self.sortedDateKeys.lastObject];
-            
-            NSDateComponents *firstComponents = [[NSCalendar currentCalendar] components:NSCalendarUnitDay | NSCalendarUnitMonth | NSCalendarUnitYear fromDate:firstDate];
-            
+            //right now last Date is same as first Date...
+            NSLog(@"Last date is %@", lastDate.description);
+            //sets calendar day to current date
+            NSCalendar *mycal = [NSCalendar currentCalendar];
+            NSDateComponents *firstComponents = [mycal components:NSCalendarUnitDay | NSCalendarUnitMonth | NSCalendarUnitYear fromDate:[NSDate date]];
+            //sets first year/month/day to appear on calendar to day of first event
             NSInteger firstYear = [firstComponents year];
             NSInteger firstMonth = [firstComponents month];
             NSInteger firstDay = [firstComponents day];
+            NSLog(@"%d is the first year", firstYear);
+            NSLog(@"%d is the first month", firstMonth);
+            NSLog(@"%d is the first day", firstDay);
             
-            NSDateComponents *lastComponents = [[NSCalendar currentCalendar] components:NSCalendarUnitDay | NSCalendarUnitMonth | NSCalendarUnitYear fromDate:lastDate];
-            NSInteger lastYear = [lastComponents year];
-            NSInteger lastMonth = [lastComponents month];
-            NSInteger lastDay = [lastComponents day];
+            //NSDateComponents *lastComponents = [[NSCalendar currentCalendar] components:NSCalendarUnitDay | NSCalendarUnitMonth | NSCalendarUnitYear fromDate:lastDate];
             
-            [self.dayPicker setStartDate:[NSDate dateFromDay:firstDay month:firstMonth year:firstYear] endDate:[NSDate dateFromDay:lastDay month:lastMonth year:lastYear]];
+            //            NSCalendar *gregorian = [[NSCalendar alloc] initWithCalendarIdentifier:NSGregorianCalendar];
```

Comment:
umm... what are these...?

Reply:
Ahh sorry... at first I was figuring out what the bug was and I wanted to see if it was issue with the calendar.